Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts

```diff
@@ -534,8 +534,13 @@ export class ResourceGroup {
                 armClient.deployments.createOrUpdate(this.taskParameters.deploymentName, deployment, (error, result, request, response) => {
                     if (error) {
                         if(error.code == "ResourceGroupNotFound" && retryCount > 0){
-                            return this.waitAndPerformAzureDeployment(armClient, deployment, retryCount);
+                            return this.waitAndPerformAzureDeployment(armClient, deployment, retryCount, spnName);
+                        }
+                        
+                        if(error.statusCode == 403) {
+                            tl.error(tl.loc("ServicePrincipalRoleAssignmentDetails", spnName, this.taskParameters.resourceGroupName));
```

Comment:
why pass it as an argument through more than one functions? better to just call this.getServicePrincipalName() here.

Reply:
To do that we'll have to make the callback Async, which won't work